src/screens/Settings/SettingsScreen.tsx

```diff
@@ -307,7 +308,9 @@ const SettingsScreen = ({ navigation, ...props }: ScreenProps) => {
           isOpen={isWalletDeleteModalOpen}
           onClose={() => setIsWalletDeleteModalOpen(false)}
           maximisedContent={Platform.OS === 'ios'}
-          Content={(props) => <WalletDeleteModal {...props} />}
+          Content={(props) => (
+            <WalletDeleteModal onDelete={() => resetNavigation(navigation, 'LandingScreen')} {...props} />
+          )}
```

Comment:
This is the best example why I am passing the `navigation` object as a prop to the `resetNavigation` function. The `WalletDeleteModal` does not have access to the `navigation` object (it's not a screen). If we'd use `useNavigate` to get the `navigation` object in the modal, TypeScript wouldn't complain and compilation would pass. If we'd try to mount the component though, we'd get an exception. Always requiring a valid `navigation` object ensures that we'll never get to that state.